data_structures/arrays/pairs_with_given_sum.py

```diff
@@ -0,0 +1,48 @@
+"""
+Author  : Siddharth Warrier
+Date    : October 3, 2023
+
+Task:
+Count the no of pairs in a given array with given sum
+
+Implementation notes: Using hashing
+The idea is that we hash the array in a dictionary
+Then go through the elements of the array
+We subtract this with the given sum
+and check if that is there in the array
+We also check the edge cases like if there are multiple same elements
+Finally we divide the count by 2
+to avoid the same pair getting counted twice
+"""
+
+
+def pairs_with_sum(arr: list, req_sum: int) -> int:
+    """
+    Return the no. of pairs with sum "sum"
+
+    >>> pairs_with_sum([1,5,7,1],6)
+    2
+    >>> pairs_with_sum([1,1,1,1,1,1,1,1],2)
+    28
+    >>> pairs_with_sum([1,7,6,2,5,4,3,1,9,8],7)
+    4
+    """
+    d: dict = {}
```

Comment:
```suggestion
    d = {}
```
This type is inferred. Also try to use more descriptive variable names

Reply:
So I had to add that because it wasn't passing all the three tasks. I know we don't have to mention the type but I did so just to pass all the three tasks

Reply:
It gave some type not specified error